cinema/tests/test_movie_viewset.py

```diff
@@ -0,0 +1,139 @@
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(len(response.data), 2)
+
+    def test_filter_movies_by_title(self):
+        url = reverse("cinema:movie-list")
+        response = self.client.get(url, {"title": "Movie One"})
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(len(response.data), 1)
+        self.assertEqual(response.data[0]["title"], "Movie One")
+
+    def test_filter_movies_by_genres(self):
+        url = reverse("cinema:movie-list")
+        response = self.client.get(url, {"genres": f"{self.genre1.id}"})
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(len(response.data), 1)
+        self.assertEqual(response.data[0]["title"], "Movie One")
+
+    def test_filter_movies_by_actors(self):
+        url = reverse("cinema:movie-list")
+        response = self.client.get(url, {"actors": f"{self.actor1.id}"})
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(len(response.data), 1)
+        self.assertEqual(response.data[0]["title"], "Movie One")
+
+    def test_create_movie_as_authenticated_user(self):
+        url = reverse("cinema:movie-list")
+        data = {
+            "title": "New Movie",
+            "description": "A new movie description",
+            "duration": 90,
+            "genres": [self.genre1.id],
+            "actors": [self.actor1.id, self.actor2.id],
+        }
+        response = self.client.post(url, data, format="json")
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+    def test_create_movie_as_admin(self):
+        self.client.credentials(HTTP_AUTHORIZATION=f"Token {self.admin_token.key}")
+        url = reverse("cinema:movie-list")
+        data = {
+            "title": "New Movie",
+            "description": "A new movie description",
+            "duration": 90,
+            "genres": [self.genre1.id],
+            "actors": [self.actor1.id, self.actor2.id],
+        }
+        response = self.client.post(url, data, format="json")
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+        self.assertEqual(response.data["title"], "New Movie")
+
+    def test_retrieve_movie(self):
+        url = reverse("cinema:movie-detail", args=[self.movie1.id])
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.data["title"], "Movie One")
+
+    def test_upload_movie_image_as_authenticated_user(self):
+        url = reverse(
+            "cinema:movie-upload-image", args=[self.movie1.id]
+        )  # Updated URL name
+        image_file = generate_image_file()
+        response = self.client.post(url, {"image": image_file}, format="multipart")
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
```

Comment:
In this test, you are testing that an authenticated user cannot upload a movie image. It would be a good idea to also test that an admin user can upload a movie image.